streams/src/main/java/org/apache/kafka/streams/kstream/internals/InternalStreamsBuilder.java

```diff
@@ -292,7 +292,9 @@ public void buildAndOptimizeTopology(final Properties props) {
 
     private void maybePerformOptimizations(final Properties props) {
 
-        if (props != null && props.getProperty(StreamsConfig.TOPOLOGY_OPTIMIZATION).equals(StreamsConfig.OPTIMIZE)) {
+        final StreamsConfig config = props != null ? new StreamsConfig(props) : null;
```

Comment:
Do you want to prevent NPE? Maybe easier to rewrite to
```
if (props != null && StreamsConfig.OPTIMIZE.equals(props.getProperty(StreamsConfig.TOPOLOGY_OPTIMIZATION))) {
```

Reply:
Yes.  But I'm thinking long term we want a ref to `StreamsConfig` available for optimizations.  But I'm not committed to doing it this way, so I'll change to what you suggested.